recipes/lammps/meta.yaml

```diff
@@ -0,0 +1,56 @@
+{% set name = "lammps" %}
+{% set version = "2018.03.16" %}
+{% set sha256 = "03f91d69f472c37a4008611dee13fad91b6bedc66dbf0393416cdacc751b57ff" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: stable_16Mar2018.tar.gz
+  url: https://github.com/lammps/lammps/archive/stable_16Mar2018.tar.gz
```

Comment:
As is, it's really easy to just change `version` at the top and not realize you're actually building the old version. It's also not very friendly with @regro-cf-autotick-bot as-is.

This way should solve the not-actually-updating issue, and possibly help with bot updates in the future (though that depends on https://github.com/regro/cf-scripts/issues/145):

```
{% set version = "2018.03.16" %}
{% set date = datetime.date.strptime(version, "%Y.%m.%d") %}
[...]

source:
  fn: {{ name }}-{{ version }}.tar.gz
  url: https://github.com/lammps/lammps/archive/stable_{{ date.day }}{{ "{:%b%Y}".format(date) }}.tar.gz
```

Unfortunately Python doesn't have a datestring code for a non-zero-padded day, so we have to do the `{{ date.day }}` thing separately.

(Note that the `fn` key is confusingly named: it's not the last component of the url, but the filename name that conda should use to cache the source under. You almost always want to leave it as `{{ name }}-{{ version }}.tar.gz`.)

Reply:
Would it be possible to go the other way with `version` being in the upstream version format? This way the bot would be happy and provide the correct version. I would make a second variable `conda_version` which was the version which is parsed from `version` and is conda complaint.

Reply:
Sure, that's easy too:

```
{% set version = "stable_16Mar2018" %}
{% set date = datetime.datetime.strptime(version.split('_')[1], "%d%b%Y") %}
{% set conda_version = "{:%Y.%m.%d}".format(date) %}

package:
  name: {{ name|lower }}
  version: {{ conda_version }}

source:
  fn: {{ name }}-{{ conda_version }}.tar.gz
  url: https://github.com/lammps/lammps/archive/{{ version }}.tar.gz
```

Does the bot actually read the current version from the graph from the `set version` line and not from `package/version`?

Reply:
But does this mean then that the PRs will be appropriate, but it'll either always or never send them based on how it sorts `stable_16Mar2018` against `2018.03.16`?

I guess it only sends one PR per version anyway, so it'll be reasonably okay?

Reply:
The idea is that we can launder the upstream version through the yaml so we can load the properly formatted version for comparison to the current version.

Reply:
Since that code hasn't been written yet it might be best to blacklist this from the bot for now, that way we don't bombard the maintainers with PRs.